mdtraj/formats/xyzfile.py

```diff
@@ -383,4 +384,12 @@ def tell(self):
 
     def __len__(self):
         """Number of frames in the file. """
-        raise NotImplementedError()
+        if str(self._mode) != 'r':
+            raise NotImplementedError('len() only available in mode="r" currently')
+        if not self._is_open:
```

Comment:
If you're opening a new handle, why do require the parsers file handle to be opened?